src/vs/workbench/api/common/extHostLanguageModelTools.ts

```diff
@@ -138,7 +138,18 @@ export class ExtHostLanguageModelTools implements ExtHostLanguageModelToolsShape
 			};
 		}
 
-		const extensionResult = await raceCancellation(Promise.resolve(item.tool.invoke(options, token)), token);
+		const progress: vscode.Progress<{ message?: string; increment?: number }> = {
+			report: value => {
+				checkProposedApiEnabled(item.extension, 'toolProgress');
+				this._proxy.$acceptToolProgress(dto.chatRequestId, dto.callId, {
+					message: value.message,
+					increment: value.increment,
+					total: 100,
+				});
+			}
+		};
+
+		const extensionResult = await raceCancellation(Promise.resolve(item.tool.invoke(options, token, progress)), token);
```

Comment:
Also don't pass progress in at all if the proposed API check failed

Reply:
A little potato potato, if they don't have the proposed API they won't get a 3rd argument in their typings

Reply:
well, typing are just smoke and mirror - folks can implement extension without dts-files and pure JS